application/src/main/java/org/thingsboard/server/service/transport/DefaultTransportApiService.java

```diff
@@ -226,6 +242,52 @@ private ListenableFuture<TransportApiResponseMsg> validateCredentials(TransportP
         }
     }
 
+    protected ListenableFuture<TransportApiResponseMsg> validateOrCreateDeviceX509Certificate(String certChain, DeviceCredentialsType credentialsType) {
+        List<String> chain = convertX509CertificateChainToList(certChain);
+        String updateDeviceCertificateValue = chain.get(0);
+        String updateDeviceCertificateHash = EncryptionUtil.getSha3Hash(updateDeviceCertificateValue);
+        String deviceCommonName = "";
+        try {
+            deviceCommonName = SslUtil.parseCommonName(readCertFile(chain.get(0)));
+        } catch (Exception ignored) {
+        }
+        for (String certificateValue : chain) {
+            String certificateHash = EncryptionUtil.getSha3Hash(certificateValue);
+            DeviceCredentials credentials = deviceCredentialsService.findDeviceCredentialsByCredentialsId(certificateHash);
+            if (credentials != null && credentials.getCredentialsType() == credentialsType) {
+                return getDeviceInfo(credentials);
+            }
+            DeviceProfile deviceProfile = deviceProfileService.findDeviceProfileByCertificateHash(certificateHash);
+            if (deviceProfile != null) {
+                X509CertificateChainProvisionConfiguration x509Configuration = new X509CertificateChainProvisionConfiguration();
+                if (deviceProfile.getProfileData().getProvisionConfiguration() instanceof X509CertificateChainProvisionConfiguration) {
```

Comment:
please add warn message if certificate is not X509CertificateChainProvisionConfiguration